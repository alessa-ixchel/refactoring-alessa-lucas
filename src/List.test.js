import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';

// this is the test case
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');
  const arr = [{id:5, title:'Test', content:'Words'}];
  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<List header="list" cards={arr} />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});