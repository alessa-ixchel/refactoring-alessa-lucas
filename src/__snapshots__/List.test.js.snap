// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Test
      </h3>
      <p>
        Words
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
